### DISCUSSION TITLE
Different area on two sides of a double-acting cylinder

### DISCUSSION THREAD

**Discussion by rohit-kumar-j** [google-deepmind/mujoco#417] [Asking for Help] [answered]
In [actuator/cylinder](https://mujoco.readthedocs.io/en/latest/XMLreference.html?highlight=intvelocity#actuator-cylinder), is the area(gain) of the cylinder considered the same on both sides of the cylinder?

If it is a double-acting cylinder with different areas on both sides like in the image below, is there an inherent way we can model this in mujoco? (different gain based on the direction of pressure(ctrl) application)

![image](https://user-images.githubusercontent.com/37873142/182597585-3f4c937f-8290-43ca-af07-55dea7ec5832.png)

Example XML
```xml
<!--Considering the same area on both sides-->
<cylinder name="act_sliding_jnt" joint="sliding_jnt" diameter="25" ctrlrange="-1.0 1.0" ctrllimited="true" />
```

**Comment by yuvaltassa** [comment, marked as answer]
By double-acting do you mean two independently controlled valves on either side of the piston?

If yes then the answer is simple: connect two actuators to the same transmission target with different areas and opposite gains (one should have negative diameter). Note that the cylinder shortcut is very simple ([implementation](https://github.com/deepmind/mujoco/blob/main/src/xml/xml_native_reader.cc#L1554)), you can just implement it with a general actuator.
